store/datastore/utils.go

```diff
@@ -38,34 +39,33 @@ func rebind(query string) string {
 
 // helper function that converts a simple repsitory list
 // to a sql IN statment.
-func toList(listof []*model.RepoLite) (string, []interface{}) {
-	var size = len(listof)
-	if size > 999 {
-		size = 999
-		listof = listof[:999]
-	}
-	var qs = make([]string, size, size)
-	var in = make([]interface{}, size, size)
-	for i, repo := range listof {
-		qs[i] = "?"
-		in[i] = repo.FullName
-	}
-	return strings.Join(qs, ","), in
-}
+func toList(listof []*model.RepoLite) ([]string, [][]interface{}) {
+	var total = len(listof)
+	var limit = 999
+	var pagesCount = int(math.Ceil(float64(total) / float64(limit)))
 
-// helper function that converts a simple repository list
-// to a sql IN statement compatible with postgres.
-func toListPosgres(listof []*model.RepoLite) (string, []interface{}) {
-	var size = len(listof)
-	if size > 999 {
-		size = 999
-		listof = listof[:999]
-	}
-	var qs = make([]string, size, size)
-	var in = make([]interface{}, size, size)
-	for i, repo := range listof {
-		qs[i] = "$" + strconv.Itoa(i+1)
-		in[i] = repo.FullName
+	var qs = make([]string, pagesCount, pagesCount)
+	var in = make([][]interface{}, pagesCount, pagesCount)
+
+	for page := 0; page < int(pagesCount); page++ {
+		var start = page * limit
+		var end = (page * limit) + limit
+		if end > total{
+			end = total
+		}
+		var size = end - start
+		var _qs = make([]string, size, size)
+		var _in = make([]interface{}, size, size)
+		for i, repo := range listof[start:end] {
+			if meddler.Default == meddler.PostgreSQL {
```

Comment:
this seems a bit inefficient since it could add thousands of extra evaluations depending on the size of the list. This was the benefit of having this split into two functions before. The prior implementation was more efficient at the expense of being dry (which was on purpose)